File: src/main/java/com/trabalho_testes/trabalho_testes/entidade/SalarioBase.java
package com.trabalho_testes.trabalho_testes.entidade;


import lombok.Data;

import javax.persistence.*;
import java.math.BigDecimal;

@Entity
@Data
public class SalarioBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private BigDecimal salario;

    @OneToOne
    private Jogador jogador;
}

File: frontend/src/app/app/servico/salario-base/salario-base.ts
import * as moment from 'moment';
import { Jogador, IJogador } from '../jogador/jogador';

export class ISalarioBase {
    id: number;
    salario: number;
    jogador: IJogador;
}

export class SalarioBase {
    id: number;
    salario: number;
    jogador: IJogador;
}